Adding bower.prune module function and bower.pruned state function

joe-niland opened the pull request:
### What does this PR do?

Expose bower's prune command in a Salt state and module, which removes those local packages which are unreferenced in the bower.json file.
Also did a small amount of refactoring, adding a function to construct the bower cli string.

### Tests written?

No

joe-niland committed:
adding bower.prune module function and bower.pruned state function

salt-jenkins commented:
@joe-niland, thanks for your PR! By analyzing the history of the files in this pull request, we identified @aspyatkin and @terminalmage to be potential reviewers.

terminalmage committed:
Pass commands as list of args

This should future-proof the code against future changes WRT ``python_shell=False``

terminalmage reviewed (changes requested):
There's only one issue that I have with this, and that is the fact that the bower module is both using ``python_shell=False`` _and_ passing the command as a string instead of a list of arguments. Using ``python_shell=False`` implies that the commands are being passed as a list of args (i.e. ``subprocess.Popen()`` with ``shell=False`` and the command passed as a list).

It's possible that we're doing some further checking when ``python_shell=False`` and splitting a string into a list of args, but that may change at a future date. Given that possibility, it's a good idea to future-proof this code and pass the commands as a list of args.

I opened https://github.com/joe-niland/salt/pull/1 to make the needed changes, can you merge that PR and then test on your end?

joe-niland committed:
Merge pull request #1 from terminalmage/pr-39628

Pass commands as list of args
